src/core/radio/trel_link.cpp

```diff
@@ -412,6 +449,8 @@ void Link::HandleAck(Packet &aAckPacket)
 
     } while (ackError == kErrorNoAck);
 
+    CheckPeerAddrOnRxSuccess(kDisallowPeerSockAddrUpdate);
```

Comment:
can you please help clarify why kDisallowPeerSockAddrUpdate option is used for TREL ACKs while invoking CheckPeerAddrOnRxSuccess? Is it because it is not MLE or MAC security protected?

Reply:
Exactly. The idea is to limit learning/updating of the TREL peer address for any secure received message (that passes Thread MLE or MAC layer security), though we expect any TREL packet to already use infra-link security (e.g., Wi-Fi link layer security).

The concern was that an attacker already on the infrastructure network (e.g., on Wi-Fi) could send fake TREL acknowledgments and trigger a TREL peer address change. I argued that such an attacker device would already need to be on the Wi-Fi network (on the local network) and could, therefore, already perform a similar attack by sending incorrect/fake mDNS messages (e.g., it could send an mDNS message announcing a different IPv6 address or port for a TREL host/service). mDNS does not provide any additional security beyond that provided by the local link.

I did not want to argue and push hard on this, so I went with this model in the PR, but I would like us to consider relaxing this constraint in the future (i.e., allow changes to TREL address on a TREL acknowledgment as well).